Merge in City damage report 202408

tallcoleman opened the pull request:
Script changes:

- Converted to standalone script instead of jupyter notebook and re-organized file structure
- Added new functionality to the script to only show damage reports within a certain date range. See the new README for details.

Report features:

- Includes all relevant damage reports from 2023 onwards (20 in total, 13 of them are net new compared to the previous report). All of the 20 appear to be related to post and rings, no racks.
- Report includes photos and confirmed asset IDs (to link to City open data) from surveys
 
Report changes:

- The MatchesDisplay tab now includes survey photos as well as the thumbnail map indicating the location. Photos are pulled from the `/photos` directory based on name and resized before insertion.
- New field “City_Problem_Type” indicates the main issue reported, e.g. “Loose/unsecured”, “Post with no ring”
- New fields “Check_311_Date” and “Check_311_Notes” to record whether there are 311 reports for the issue already (though the utility of this is limited, since the City 311 page only shows 311 reports from the last 30 days at most)
- New MatchesTable tab shows the report with the first (nearest or confirmed) linked data point from the City

tallcoleman committed:
Update .gitignore

tallcoleman committed:
Re-organized location of past report

tallcoleman committed:
Added photos of surveyed locations

tallcoleman committed:
Reorganized reference files

tallcoleman committed:
Converted report generation notebook into script

tallcoleman committed:
Reorganized output reports

tallcoleman committed:
Reorganized main script location and added test

tallcoleman committed:
Update test_filter_by_date.py

Auto-format update

tallcoleman committed:
Added draft documentation

tallcoleman committed:
Update generate_damage_report.py

Inserted date filtering function and added function docstrings

tallcoleman committed:
Update generate_damage_report.py

Clean up

tallcoleman committed:
Update generate_damage_report.py

refactored to put matching of city data and bikespace reports into separate function

tallcoleman committed:
Update generate_damage_report.py

Added functionality to match city data by identified asset IDs

tallcoleman committed:
Update generate_damage_report.py

Tweak to excel output

tallcoleman committed:
Added photos from survey

tallcoleman committed:
Update BikeSpace Data Notes and Cleanup - Data.csv

tallcoleman committed:
Excel report now includes survey photos

Also fixed some formatting with the Excel report

tallcoleman committed:
Update generate_damage_report.py

Updated report Notes page documentation

tallcoleman committed:
Added survey photos and notes

tallcoleman committed:
Updated documentation and added excluded status

tallcoleman committed:
Added Match table in addition to display sheet

tallcoleman committed:
Added PDF copy of report

sonalranjit commented on .gitignore:
you probably want to ignore this change until the `develop` branch is merged into `main` to avoid conflicts

tallcoleman replied:
Fixed by #146

sonalranjit reviewed (changes requested):
Thanks @tallcoleman , awesome to have everything in one place. I've just mentioned the changes for the `.gitignore` other than that we should be good to merge

tallcoleman committed:
Updated reference data and dashboard link

tallcoleman committed:
Set up uv environment

tallcoleman committed:
Updated to match new api date format

tallcoleman committed:
Ran update

tallcoleman committed:
Made PDF version of report

tallcoleman committed:
Re-organized and cleaned up project

tallcoleman commented:
Did some additional clean up of this script:

- Added [uv](https://docs.astral.sh/uv/) as an python/dependencies manager
- Separated out tests folder
- Updated date parsing to match current api format (see: #184 )
- Updated dashboard link generation to point to new dashboard

sonalranjit reviewed (approved):
Coool! thanks @tallcoleman
